src/test/scala/sigmastate/utxo/CollectionOperationsSpecification.scala

```diff
@@ -373,13 +442,31 @@ class CollectionOperationsSpecification extends SigmaTestingCommons {
         |  indexCollection.forall(elementRule)
          }""".stripMargin
 
-    val indexCollection = ConcreteCollection((0 until 6).map(i => IntConstant(i)))
-    val string = ConcreteCollection(Array(1, 1, 0, 0, 0, 1).map(i => IntConstant(i)))
-    val indexId = 21.toByte
-    val index = TaggedInt(indexId)
-    val element = ByIndex(string, If(LE(index, 0), 5, Minus(index, 1)))
-    val elementRule = BinOr(EQ(element, 0), EQ(element, 1))
-    val expectedPropTree = ForAll(indexCollection, indexId, elementRule)
+    val element = ByIndex(
+      ValUse(4, SCollectionType(SInt)),
+      If(
+        ValUse(7, SBoolean),
+        BlockValue(Vector(ValDef(8, List(), ValUse(3, SInt))), ValUse(8, SInt)).asIntValue,
+        Minus(ValUse(5, SInt), ValUse(2, SInt))),
+      None)
+val expectedPropTree = BlockValue(
+  Vector(
+    ValDef(1, List(), IntConstant(0)),
+    ValDef(2, List(), IntConstant(1)),
+    ValDef(3, List(), IntConstant(5)),
```

Comment:
@greenhat to increase effect of constant segregation we need to treat the constants specially and don't create ValDef even if the constant is used more than one time, because two equal constants don't always have the same meaning. 
This will not apply for GroupElement constants of ProveDlog, because in this case different constants have different meaning, thus it is ok for them to create ValDef

Reply:
Fixed.